### ISSUE TITLE
Twitter card does not validate (and shows no image)

### ISSUE THREAD

**Issue by aronbos** [humhub/humhub#6008]
![afbeelding](https://user-images.githubusercontent.com/48289775/210333772-19625fce-2dd7-4f2b-978b-32f9549008b8.png)

In Humhub 1.3 the twitter card displays no image.  Checking it with https://cards-dev.twitter.com/validator it does not show a preview. So a name or property is set incorrect. But which one?

**Comment by aronbos** [comment]
The OG image is generated as: <meta name="og:image" content= but the correct specification is <meta property="og:image" content=  So change name in property.

So change [protected/humhub/modules/web/pwa/widgets/LayoutHeader.php to correct the image. 

The second change would be to use OG for Twitter.  On https://developer.twitter.com/en/docs/twitter-for-websites/cards/guides/getting-started:
Note that while Open Graph recommends specifying the “og” RDFa Core 1.1 CURIE prefix mapping via <html prefix="og: http://ogp.me/ns#">, no such markup is required for Twitter cards and its use of the twitter: prefix in a HTML meta element’s name attribute. Open Graph protocol also specifies the use of property and content attributes for markup (<meta property="og:image" content="http://example.com/ogp.jpg"/>) while Twitter cards use name and content. Twitter’s parser will fall back to using property and content, so there is no need to modify existing Open Graph protocol markup if it already exists.

So remove all Twitter lines that fall back to OG to eliminate redundancy and keep the html file small.

I can make a pull request with the changes, if the Humhub team agrees

**Comment by aronbos** [comment]
See  https://twittercommunity.com/t/card-error-unable-to-render-or-no-image-read-this-first/62736 
Short: preview does not work and is abandoned.
But the image does not show up in a tweet.

**Comment by ArchBlood** [comment]
This kind of metadata is handled by `/protected/humhub/modules/ui/view/components/ViewMeta` now.

But you are correct `'property' => 'twitter:image'` can be used here. But going by Twitter's docs `'name' => ''` should be used for all Twitter metatags. :thinking:

But do keep in mind the following as well;
https://developer.twitter.com/en/docs/twitter-for-websites/cards/guides/getting-started

**Comment by aronbos** [comment]
Commenting out line 87 in viewMeta `//  $this->registerTwitterMetaTags();` still leaves a functioning Twitter. So we can remove it, I think. It is just double with OpenGraph.

I can not find the reason why the image is not displayed in Twitter. Anybody any hints?

**Comment by aronbos** [comment]
Changing line  222 in `/protected/humhub/modules/ui/view/components/ViewMeta` to
            `$this->view->registerMetaTag(['property' => 'og:image', 'content' => $this->images[0]]);`
will pass the test on [https://socialsharepreview.com](url)
Changing name to property is the solution.

**Comment by ArchBlood** [comment]
> Changing line  222 in `/protected/humhub/modules/ui/view/components/ViewMeta` to
>             `$this->view->registerMetaTag(['property' => 'og:image', 'content' => $this->images[0]]);`
> will pass the test on [https://socialsharepreview.com](url)
> Changing name to property is the solution.
> P.S. I disabled Twitter as it is double with Opengraph.
> 

I agree, it's better to switch to using `property` here, although we'll also have to add a `summary_large_image` for `twitter:card` as it's a requirement here regardless. :thinking:

**Comment by aronbos** [comment]
Why don't we remove twitter and use OpenGraph only? It works correct, gives a shorter html-page and there is less code to maintain.
